core-plugins/src/main/java/co/cask/hydrator/plugin/transform/LogParserTransform.java

```diff
@@ -102,6 +102,15 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
       LOG.error("Log format not currently supported.");
       throw new IllegalStateException("Unsupported log format: " + config.logFormat);
     }
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      if (!validateInputSchemaType(inputSchema.getType())) {
+        throw new IllegalArgumentException(
```

Comment:
probably better for the validate method to throw this, since we already have the message, and because validate is the one that knows whats invalid.